internal/service/advancedcluster/fw_data_source_advanced_cluster.go

```diff
@@ -0,0 +1,429 @@
+			Attributes: map[string]schema.Attribute{
+				"standard": schema.StringAttribute{
+					Computed: true,
+				},
+				"standard_srv": schema.StringAttribute{
+					Computed: true,
+				},
+				"private": schema.StringAttribute{
+					Computed: true,
+				},
+				"private_srv": schema.StringAttribute{
+					Computed: true,
+				},
+				"private_endpoint": schema.ListNestedAttribute{
+					Computed: true,
+					NestedObject: schema.NestedAttributeObject{
+						Attributes: map[string]schema.Attribute{
+							"connection_string": schema.StringAttribute{
+								Computed: true,
+							},
+							"endpoints": schema.ListNestedAttribute{
+								Computed: true,
+								NestedObject: schema.NestedAttributeObject{
+									Attributes: map[string]schema.Attribute{
+										"endpoint_id": schema.StringAttribute{
+											Computed: true,
+										},
+										"provider_name": schema.StringAttribute{
+											Computed: true,
+										},
+										"region": schema.StringAttribute{
+											Computed: true,
+										},
+									},
+								},
+							},
+							"srv_connection_string": schema.StringAttribute{
+								Computed: true,
+							},
+							"srv_shard_optimized_connection_string": schema.StringAttribute{
+								Computed: true,
+							},
+							"type": schema.StringAttribute{
+								Computed: true,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func advClusterDSReplicationSpecsSchemaAttr() schema.SetNestedAttribute {
+	return schema.SetNestedAttribute{
+		Computed: true,
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: map[string]schema.Attribute{
+				"id": schema.StringAttribute{
+					Computed: true,
+				},
+				"num_shards": schema.Int64Attribute{
+					Computed: true,
+				},
+				"container_id": schema.MapAttribute{
+					ElementType: types.StringType,
+					Computed:    true,
+				},
+				"zone_name": schema.StringAttribute{
+					Computed: true,
+				},
+				"region_configs": schema.SetNestedAttribute{
+					Computed: true,
+					NestedObject: schema.NestedAttributeObject{
+						Attributes: map[string]schema.Attribute{
+							"analytics_specs":        advancedClusterRegionConfigSpecsAttr(),
+							"auto_scaling":           advancedClusterRegionConfigAutoScalingSpecsAttr(),
+							"analytics_auto_scaling": advancedClusterRegionConfigAutoScalingSpecsAttr(),
+							"backing_provider_name": schema.StringAttribute{
+								Computed: true,
+							},
+							"electable_specs": advancedClusterRegionConfigSpecsAttr(),
+							"priority": schema.Int64Attribute{
+								Computed: true,
+							},
+							"provider_name": schema.StringAttribute{
+								Computed: true,
+							},
+							"read_only_specs": advancedClusterRegionConfigSpecsAttr(),
+							"region_name": schema.StringAttribute{
+								Computed: true,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func advancedClusterRegionConfigAutoScalingSpecsAttr() schema.ListNestedAttribute {
```

Comment:
advance or adv? Let's define one of these and use it everywhere
```suggestion
func advClusterRegionConfigAutoScalingSpecsAttr() schema.ListNestedAttribute {
```